### ISSUE TITLE
Contributors wanted -- SkyBlock endpoint upgrade

### ISSUE THREAD

**Issue by MaxKorlaar** [MaxKorlaar/hypixel-signatures#361]
As was announced last year, [Hypixel is deprecating their v1 SkyBlock profile endpoints](https://hypixel.net/threads/public-api-endpoint-changes.5537323/). I only discovered this now to be honest -- I didn't even know they finally implemented a developer portal. Neat! But the reason I discovered this so late is because I've been occupied with a lot of other things. I frankly don't have enough time right now to keep this website up to date.

I'm looking for people willing to help me out in switching to this endpoint -- I took a lot of code from a popular SkyBlock stats website and converted it to PHP but I'm sure it's outdated by now as well. If anyone's willing to take over the project as a whole that'd be great too.

As far as I know the popularity of the project seems to be relatively stable for the past year. Visitor numbers aren't enormous but they're consistent around 5-6k a week. I'm not actually measuring how many people use the main feature of the site (in my opinion); the signature images. Ideally I should start tracking this to see if maintaining the SkyBlock side of things is even worth it.

**Comment by 2zqa** [comment]
Consider pinning this issue, to add some extra awareness? I'm checking the repo to see if I can help in any way, if at all. I haven't touched PHP in years, although I'd be sad to see this beautiful project silently die. It seems the deadline has already passed for said endpoints.

**Comment by MaxKorlaar** [comment]
Yeah good idea! It'd be sad for the site to die but I'm quite busy :(. I'll probably be able to update the framework and all dependencies and all, but not sure if I'll have the time to fix the broken skyblock signatures now. Perhaps I'll first need to start tracking signature popularity.

**Comment by 2zqa** [comment]
So from what I understand Plancke's hypixel-php library doesn't even support the new skyblock changes? I couldn't find any issues about it though. I looked at your fork of the repository (after finding it in the lockfile) and found it was only [14 commits behind](https://github.com/MaxKorlaar/hypixel-php/compare/master...Plancke%3Ahypixel-php%3Amaster).

If so that would suck. In an ideal world we would just use an autogenerated api client from the OpenAPI spec with something like https://openapi-generator.tech/docs/generators/php/, however that would be quite a big refactor.

**Comment by MaxKorlaar** [comment]
Yeah, I would have no issues using a generated client from the API spec but the nice thing about Plancke's library is that it also includes a lot of helpers and methods specifically tailored towards the minigames or things like how names render in-game when combined with a rank tag and stuff like that! 

Skyblock is just such a huge project that while I honestly tried to mimic the popular Skycrypt website in PHP, I decided I was just not going to be able to keep up and opted to just include that whole project in here for a bunch of variables. It seems that I still fell behind so I'd either have to rewrite all that logic currently in place for SkyBlock or add a few more bindings to just directly call their logic in JavaScript and use that output in PHP!

**Comment by 2zqa** [comment]
Right I see, doing the calculations here is a non-starter... Like you mentioned, calculating is hard, and constantly maintaining it is even worse.

I noticed there is an [SkyCrypt OpenAPI spec](https://sky.shiiyu.moe/api), but it seems they don't provide calculated values in there either(?). So, the only way is to scrape their website/run their JS locally? If so, I think it's best to even just disable the signatures because that's going to be ugly.

**Comment by MaxKorlaar** [comment]
I wonder if I can utilise the current trend of AI agents and AI-written code to figure all of this data out, as I'm still low on time for this sadly

**Comment by MaxKorlaar** [comment]
> Do you still need help? Sorry for necroposting

Always welcome!
